client/tests/simulator_tests.rs

```diff
@@ -51,9 +50,18 @@ fn get_started_tpm() -> (TpmSim, TcpTpm) {
 }
 
 #[test]
-fn test_get_manufacturer_id() {
+fn test_get_capability_manufacturer_id() {
     let (_sim_lifeline, mut tpm) = get_started_tpm();
-    assert!(get_manufacturer_id(&mut tpm).is_ok());
+
+    const CMD: GetCapabilityCmd = GetCapabilityCmd {
+        capability: TPM2Cap::TPMProperties,
+        property: TPM2PT::Manufacturer,
+        property_count: 1,
+    };
+    let resp = run_command(&CMD, tpm)?;
+    assert!(let TpmsCapabilityData::TpmProperties(prop) = resp.capability_data);
```

Comment:
I don't think you can do a `let` inside an assert like this... Though I've never tried it myself, I do see that this is causing a build error for you.

Reply:
The way to do this btw is like this:

```rs
let TpmsCapabilityData::TpmProperties(prop) = resp.capability_data else { panic!("Unexpected variant data.") };
```

This is okay for testing, but you should not do such a thing in non-test code.